### DISCUSSION TITLE
Subgrouped lag transformations

### DISCUSSION THREAD

**Discussion by concimuscb** [Nixtla/mlforecast#395] [Q&A]
Is there any way to perform grouped lagged transformations?
Say for example that the unique ID is in the format cat_id@sku_id and I would like to add the expanding mean of each category to all its skus.

How could I accomplish this with MLForecast?

**Comment by jmoralez** [comment]
Hey. This is the same as what is requested in #194, right?

**Reply by concimuscb** [reply]
Hello, not sure if it is exactly the same. As far as I understood, #194 references the two "extreme" cases: transformations per unique_id and transformations for the sum of all unique_id. 

This would be some kind of middle ground where ids are first grouped and then transformations are applied on a group per group basis.

**Reply by jmoralez** [reply]
The idea of that issue is being able to provide some columns to aggregate by first and then computing the transformations on the aggregations.

**Reply by concimuscb** [reply]
Then I believe it would be addressed yes.
